### PR TITLE
Reflect recent ProviderConfig changes

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

Specifically, the change of schema introduced by crossplane/crossplane-runtime#212, and the fact that resources default to using the ProviderConfig named 'default'.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
By running through all of the examples under `docs/snippets/compose`.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: docs/introduction/managed-resources.md [hunk 1]
```diff
@@ -40,49 +41,52 @@ spec:
       name: mysql-secret
       namespace: crossplane-system
   providerConfigRef:
-    name: mycreds
+    name: default
   reclaimPolicy: Delete
```
Comment (hasheddan) [inline docs/introduction/managed-resources.md:45]:
```suggestion
  deletionPolicy: Delete
```

File: docs/introduction/composition.md [hunk 2]
```diff
@@ -437,8 +433,6 @@ spec:
             endIpAddress: 10.10.255.254
             virtualNetworkSubnetIdSelector:
               name: sample-subnet
-        providerConfigRef:
-          name: example
         reclaimPolicy: Delete
```
Comment (hasheddan) [inline docs/introduction/composition.md:436]: I believe this and other instances in this file can be removed or replaced with `deletionPolicy`

File: docs/introduction/managed-resources.md [hunk 3]
```diff
@@ -40,49 +41,52 @@ spec:
...
-  regarding authentication of Crossplane to the provider. `ProviderConfig` resources
-  refer to `Secret` and potentially contain other information regarding authentication.
+
+* `providerConfigRef`: Reference to the `ProviderConfig` resource that will
+  provide information regarding authentication of Crossplane to the provider.
+  `ProviderConfig` resources refer to `Secret` and potentially contain other
+  information regarding authentication. The `providerConfigRef` is defaulted to
+  a `ProviderConfig` named `default` if omitted.
 
 * `reclaimPolicy`: Enum to specify whether the actual cloud resource should be
```
Comment (hasheddan) [inline docs/introduction/managed-resources.md:63]: I believe this should be updated to `deletionPolicy` and possible values changed to `Delete` and `Orphan`

**Review by hasheddan — APPROVED** [review]
A few small comments, but they are not strictly related to `ProviderConfig` so feel free to go ahead and merge if outside the scope of this PR 👍

**Comment by negz** [comment]
Thanks @hasheddan - I'll follow up with another PR that addresses the reclaim vs deletion policies.

**Pull request merged**